### ISSUE TITLE
gui: develop an interface for the graphically challenged.

### ISSUE THREAD

**Issue by sahib** [sahib/rmlint#130]
The current gui implementation can be found under ``gui/`` in https://github.com/sahib/rmlint/tree/gtk-gui/gui. If someone wants to have a take on it before I have the time he could tell me to clean it up. 

It's written in Python and uses ``gtk >= 3.14``. Most of the functionality is still missing, but the core functionality, i.e. reading rmlint's json output somewhat works.

**Comment by SeeSpotRun** [comment]
To run the gui, first checkout the gtk-gui branch then:
```
[rmlint] $ sudo scons DEBUG=1 --prefix=/usr install
[rmlint] $ cd gui
[rmlint/gui] $ python3 setup.py install
[rmlint/gui] $ PYTHONPATH=. make run
```

**Comment by sahib** [comment]
Updates:

- Some beginning code cleanup.
- Architecture starts to make sense.
- Should be installable via `python setup.py install`
- New ``rmlint --gui`` option that simply launches the python script.

Current TODOs are mentioned here:

https://github.com/sahib/rmlint/blob/gtk-gui/gui/TODO

**Comment by sahib** [comment]
It's starting to get pretty and slightly useful.
News:

- Added a cmdline frontend. (```rmlint --gui --help```)
- All views (even settings) are searchable now via *Ctrl-F*.
- Default actions introduced, *Ctrl-Enter* will do the suggested action.
- Made treeview sortable (harder than it sounds...)
- Results can be filtered and only a partial script can be generated (through ``--replay``)

Note: No files will be harmed currently, the `run script` always uses dry run.

**Comment by sahib** [comment]
It should be suitable for some rough testing now. 
Bugs should be expected, but I already added a lot of polishing in the last few days.
Some friends tried (and struggled) to use it, so I changed some details.

Notable changes (beside lots of small bugfixes):

- Search can filter by size/count. Later perhaps also per mtime.
- Redesign views to use GtkPaned where necessary (more user freedom)
- Should look/feel generally smoother now.
